File: 151-reverse-words-in-a-string/reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        string result;
        stack<string> st;
        stringstream ss(s);
        string t;

        while(ss >> t){
            st.push(t);
        }

        while(!st.empty()){
            result +=st.top();
            st.pop();
            if(!st.empty()){
                result+= ' ';
            }
        }
    return result;
    }
};

File: 151-reverse-words-in-a-string/reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {

        StringBuffer word = new StringBuffer();
        StringBuffer result = new StringBuffer();

        for (int i = s.length() - 1; i >= 0; i--) {
            if (s.charAt(i) != ' ') {
                word.append(s.charAt(i));
            } else if (word.length() > 0) {
                word.reverse();
                result.append(word).append(' ');
                word.setLength(0);
            }
        }

        if (word.length() > 0) {
            word.reverse();
            result.append(word);
        } else if (result.length() > 0) {
            result.setLength(result.length() - 1);
        }
        return result.toString();
    }
}